zmq.hpp

```diff
@@ -32,20 +32,34 @@
 #endif
 #endif
 
+// included here for _HAS_CXX* macros
+#include <zmq.h>
+
+#if defined(_MSCV_LANG)
+#define CPPZMQ_LANG _MSCV_LANG
+#elif defined(__cplusplus)
+#define CPPZMQ_LANG __cplusplus
+#else
+#define CPPZMQ_LANG 0
```

Comment:
I think this should rather be some `#error`. If `__cplusplus` is not defined, we are not compiling in C++ mode, and that can't work.